### PR TITLE
3189 fix group renaming

### PR DESCRIPTION
**Pull request by Ali96kz**
fix group renaming #3189 .

### CODE REVIEW

File: src/main/java/org/jabref/gui/groups/GroupNodeViewModel.java [hunk 1]
```diff
@@ -166,7 +174,7 @@ public boolean equals(Object o) {
     @Override
     public String toString() {
         return "GroupNodeViewModel{" +
-                "displayName='" + displayName + '\'' +
+                "displayNameProperty='" + displayNameProperty + '\'' +
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/gui/groups/GroupNodeViewModel.java]: _You should probably call here .getValue()

Comment (Ali96kz) [reply src/main/java/org/jabref/gui/groups/GroupNodeViewModel.java]: fixed

**Comment by Ali96kz** [comment]
Fixed.

**Comment by Siedlerchr** [comment]
Please have a look at the failing tests and please test if the latex stuff is still transformed to Unicode after rename

File: src/main/java/org/jabref/gui/groups/GroupNodeViewModel.java [hunk 2]
```diff
@@ -64,7 +66,9 @@ public GroupNodeViewModel(BibDatabaseContext databaseContext, StateManager state
         this.localDragBoard = Objects.requireNonNull(localDragBoard);
 
         LatexToUnicodeFormatter formatter = new LatexToUnicodeFormatter();
-        displayName = formatter.format(groupNode.getName());
+        displayNameProperty = new SimpleStringProperty(formatter.format(groupNode.getGroup().getName()));
```
Comment (tobiasdiez) [inline src/main/java/org/jabref/gui/groups/GroupNodeViewModel.java]: This should be
```suggestion
        displayNameProperty = new SimpleStringProperty();
        displayNameProperty.bind(Easybind.map(groupNode.getGroup().nameProperty(), formatter:format));
```
We want to update the displayed name every time the name of the group changes, using the formatter.

File: src/main/java/org/jabref/gui/groups/GroupNodeViewModel.java [hunk 3]
```diff
@@ -158,15 +166,15 @@ public boolean equals(Object o) {
             return false;
         }
 
-        GroupNodeViewModel that = (GroupNodeViewModel) o;
+            GroupNodeViewModel that = (GroupNodeViewModel) o;
```
Comment (tobiasdiez) [inline src/main/java/org/jabref/gui/groups/GroupNodeViewModel.java]: The indent looks to be off.

File: src/main/java/org/jabref/gui/groups/GroupTreeViewModel.java [hunk 4]
```diff
@@ -161,20 +161,6 @@ public void editGroup(GroupNodeViewModel oldGroup) {
         Optional<AbstractGroup> newGroup = dialogService
                 .showCustomDialogAndWait(new GroupDialog(oldGroup.getGroupNode().getGroup()));
         newGroup.ifPresent(group -> {
-            // TODO: Keep assignments
```
Comment (tobiasdiez) [inline src/main/java/org/jabref/gui/groups/GroupTreeViewModel.java]: Why did you removed this code?

File: src/main/java/org/jabref/gui/util/ViewModelTreeTableCellFactory.java [hunk 5]
```diff
@@ -19,12 +20,12 @@
  */
 public class ViewModelTreeTableCellFactory<S, T> implements Callback<TreeTableColumn<S, T>, TreeTableCell<S, T>> {
 
-    private Callback<S, String> toText;
+    private Callback<S, StringProperty> toText;
     private Callback<S, Node> toGraphic;
     private Callback<S, EventHandler<? super MouseEvent>> toOnMouseClickedEvent;
     private Callback<S, String> toTooltip;
 
-    public ViewModelTreeTableCellFactory<S, T> withText(Callback<S, String> toText) {
```
Comment (tobiasdiez) [inline src/main/java/org/jabref/gui/util/ViewModelTreeTableCellFactory.java]: Changing this to `StringProperty` shouldn't be necessary. Normally, the table automatically refreshes the cell if the underlying data is changed. Please revert this change even if then the bug is no longer fixed (see the general comment).

**Comment by tobiasdiez** [comment]
@Ali96kz why did you close the pull request? You can simply push your changes to your branch and the PR is updated.

**Comment by Ali96kz** [comment]
@tobiasdiez It's create a problem with a group adding.

**Comment by Ali96kz** [comment]
@tobiasdiez Could you review again?
